src/solve/solver.rs

```diff
@@ -183,12 +183,14 @@ impl Solver {
             match self.cycle_strategy {
                 CycleStrategy::Tabling if slot.cycle => {
                     let actual_answer = result.as_ref().ok().map(|s| s.clone());
-                    if actual_answer == answer {
-                        // Fixed point: break
-                        return result;
-                    } else {
-                        answer = actual_answer;
-                    }
+                    let fixed_point = actual_answer == answer;
```

Comment:
I suspect it as well although ambig with guidance can perform some unification, but I cannot find examples where a cycle gives ambig with guidance after being fed. Nonetheless stopping when we get `Ambig` regardless of the guidance should not affect correctness of the final answer.